src/main/java/ca/concordia/soen6441/project/gameplay/orders/Bomb.java

```diff
@@ -1,11 +1,72 @@
 package ca.concordia.soen6441.project.gameplay.orders;
 
+import ca.concordia.soen6441.project.interfaces.Country;
 import ca.concordia.soen6441.project.interfaces.Order;
+import ca.concordia.soen6441.project.interfaces.Player;
+import ca.concordia.soen6441.project.log.LogEntryBuffer;
 
+/**
+ * Implementation of the Order interface.
+ * Represents a bomb card in the game with its properties and behaviors.
+ */
 public class Bomb implements Order {
+
+    private final Country d_targetTerritory;
+    private final Player d_initiator;
+
+    public Bomb(Player p_initiator, Country p_targetTerritory) {
+        this.d_targetTerritory = p_targetTerritory;
+        this.d_initiator = p_initiator;
+    }
+
+    /**
+     * Execute the order to bomb an enemy country
+     */
     @Override
     public void execute() {
-        // TODO #67
 
+        String l_errorMsg = null;
+        if ((l_errorMsg = validate()) != null)
+        {
+            // Error found, write to screen and add to log
+            LogEntryBuffer.getInstance().appendToBuffer(l_errorMsg, true);
+        }
+        else {
+            d_targetTerritory.setTroops((int)(Math.floor(d_targetTerritory.getTroops()/2.0)));

```

Comment:
LogEntryBuffer "True" statement required here.
Detail:
- that a bomb went off at <location>
- original # of troops is now #

Reply:
<Player> just dropped a bomb in <>!  <Opponent>'s soldiers fell from <original #> to <new #>!